### ISSUE TITLE
LMC on tray icon act like RMC

### ISSUE THREAD

**Issue by hxss** [telegramdesktop/tdesktop#4059]
### Steps to reproduce
1. Close Telegram window to tray
2. Left click on Telegram tray icon

### Expected behaviour
Telegram window opens up.

### Actual behaviour
The Menu pops up as if right clicked.

### Configuration
**Operating system:**
4.13.11-1-ARCH x86_64

**Version of Telegram Desktop:**
1.1.23 Installed from [AUR](https://aur.archlinux.org/packages/telegram-desktop/).

</details>

**Comment by stek29** [comment]
Please try executable from desktop.telegram.org (you can install it from AUR: telegram-desktop-bin)

**Comment by hxss** [comment]
Failure while downloading https://updates.tdesktop.com/tlinux/tsetup.1.1.23.tar.xz

**Comment by hxss** [comment]
@stek29 same behavior with telegram-desktop-bin

**Comment by stek29** [comment]
What DE do you use?

**Comment by hxss** [comment]
XFCE 4.12

**Comment by haevalencia** [comment]
But that's the normal behavior of the status icon. Another thing would be if you are using a DE with support for AppIndicator or KStatusNotifier Item.

**Comment by hxss** [comment]
@haevalencia this is not a normal behavior. This is first app in my tray, that show menu on left click. All other apps toggle they window(xfce4-terminal, deluge, blueberry, vlc(qt), keepassxc(qt)).

**Comment by haevalencia** [comment]
@hxss I will be clearer. This is the _normal behavior_ for the Telegram Desktop status icon.
One of the reasons for GNOME developers to want to remove support from the status icon is that all applications implement it in a different way and therefore behave in a different way.

**Comment by Felixoid** [comment]
What about making this behavior changeable by settings?

**Comment by Felixoid** [comment]
Isn't it issue related to #3175?

**Comment by hxss** [comment]
@Felixoid Yes, related. This is about one problem. But #3175 was closed without answer or solution published.

**Comment by auchri** [comment]
@hxss Really?

> john-preston closed this issue in c1aa1c5 9 months ago

**Comment by hxss** [comment]
@auchri What do you mean?

**Comment by Felixoid** [comment]
I guess, @hxss means, that issue closed with solution

**Comment by hxss** [comment]
What solution are you talking about? What I need to do to solve this problem? I'm now use v1.2.1 and LMC still open menu instead toggle window.

**Comment by Felixoid** [comment]
I'm sorry, @hxss, it was the wrong mention, I wanted to write "@auchri" =)
And yes, LMC is still working wrong.

**Comment by hxss** [comment]
I'm have just upgraded telegram-desktop (1.2.6-1 -> 1.2.6-3). Finally, the LMC works as it should.
It's so cool to open and close the telegram window with just one click!
And tray menu now use another fonts and styles than system(not a problem for me 🙂).
![screenshot from 2018-01-10 14-26-32](https://user-images.githubusercontent.com/17298149/34765165-9bb7a664-f612-11e7-89ed-a6aa201a2bfc.png)

**Comment by Felixoid** [comment]
```
pacman -Qi telegram-desktop 
Name            : telegram-desktop
Version         : 1.2.8-1
```

Sequel: RMC returned

**Comment by stek29** [comment]
That's not official package. Try telegram-desktop-bin from AUR.

**Comment by Felixoid** [comment]
The package from the community repo isn't official?
```
pacman -Ss telegram-desktop
community/telegram-desktop 1.2.8-1 [installed]
    Official Telegram Desktop client
```
https://www.archlinux.org/packages/community/x86_64/telegram-desktop/

**Comment by DrakeTDL** [comment]
IDK about that link but you can get the official version from [Github Releases](https://github.com/telegramdesktop/tdesktop/releases) or from [Telegram's website](https://desktop.telegram.org)

**Comment by Felixoid** [comment]
either of `telegram-desktop` or `telegram-desktop-bin` work with LMC as RMC
```
pacman -Qi telegram-desktop-bin
Name            : telegram-desktop-bin
Version         : 1.2.6-1
```
@DrakeTDL there is source of [package](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/telegram-desktop#n18), exactly from github releases, and there are [patches](https://git.archlinux.org/svntogit/community.git/tree/trunk?h=packages/telegram-desktop) applied while building.
Maybe [this one](https://git.archlinux.org/svntogit/community.git/tree/trunk/no-gtk2.patch?h=packages/telegram-desktop) broke LMC?

**Comment by hxss** [comment]
@Felixoid in 1.2.6-4 gtk3 libs [was returned](https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/telegram-desktop&id=3f47bddefd3b06f0c816c1f88ad78318d2c11763), so we have again menu by LMC. But when you use 1.2.6-3 or official release - you [can't use native Linux notifications](https://github.com/telegramdesktop/tdesktop/issues/3831#issuecomment-356715454).

**Comment by john-preston** [comment]
@Winterwolf Does haguchi show an icon using that lib? How does its icon handle clicks!

**Comment by Winterwolf** [comment]
@john-preston yes, it also show context menu on left click.

**Comment by john-preston** [comment]
@Winterwolf Well, tdesktop uses this library as well, so it behaves the same way.

There are several ways tray icon is supported in tdesktop and it is near to impossible to choose the right way in all cases of this huge zoo of Linux DEs and installed libraries.

Currently there are several conditions based on env variables that guide which implementation to prefer, which not to use because it was not working at all in some cases. And libappindicator at least works when it is available, so most of the times it is preferred.

**Comment by ztefn** [comment]
@Winterwolf @john-preston Haguichi [listens to scroll events on the indicator](https://github.com/ztefn/haguichi/blob/master/src/indicator.vala#L44), hiding the window when you scroll down and showing it again when scrolling up. It's a bit of a hack around the appindicator limitations in order to keep providing an option to quickly show/hide the window, like you could by clicking on status icons in the old days.

Maybe this would be an option for tdesktop as well. However, not all desktop environments support scroll events on appindicators, and I'm affraid XFCE is one of them...

**Comment by hxss** [comment]
telegram-desktop 1.2.24-1 works perfect for me now on Arch Linux XFCE! Both tray mouse events and system notifications work!

**Comment by stek29** [comment]
@hxss because it doesn’t use libappindicator I guess :(

**Comment by hxss** [comment]
@stek29 you right. Dont know what is it and for what this pkg. I'm happy now. Only 1 click on tray icon and tg opens/close!

**Comment by eMPee584** [comment]
Another step that would seriously decrease the annoyance is to reverse order in this context menu.. right now, if you double click on the tray icon, it opens the menu and selects **Quit Telegram**.

**Comment by hxss** [comment]
on fedora 28 telegram-desktop-1.4.3-1.fc28.x86_64 still show menu on LMC. Libappindicator installed but tg doesn't depend from it.

**Comment by ncc-1701-a** [comment]
The same problem on Debian sid (unstable) with XFCE 4.14.
Telegram v.1.8.15.
Other applications (Skype (electron based), Viber (qt5), uget (GTK2) and deluge (GTK3)) are have not this issue.
Additional information: appindicator (libappindicator3-1) has been purged deliberately: it does not help.
Please fix this.

**Comment by ncc-1701-a** [comment]
More information: when I launch the Telegram with command "env XDG_CURRENT_DESKTOP=NONE telegram-desktop -startintray -- %u", LMC works as expected.

**Comment by ncc-1701-a** [comment]
I've made more tests for XFCE DE and Telegram.
I've tested Telegram with different XDG_CURRENT_DESKTOP variables:

$ telegram-desktop
proto: D-BUS, LMC act like RMC;

$ env XDG_CURRENT_DESKTOP=NONE telegram-desktop
proto: XEMBED, LMC act properly;

$ env XDG_CURRENT_DESKTOP=KDE telegram-desktop
proto: D-BUS, LMC act properly;

$ env XDG_CURRENT_DESKTOP=GNOME telegram-desktop
proto: D-BUS, LMC act like RMC.

So, I almost completely sure, that it is Telegram bug (not XFCE bug).
